### PR TITLE
Fix GCS Mock Behavior for Missing Bucket

### PR DESCRIPTION
**Pull request by original-brownbear**
We were throwing a 500 instead of a 404 for a missing bucket.
This would make yaml tests needlessly wait for multiple seconds, retrying
the 500 response with backoff, in the test checking behavior for missing buckets.

Also fixed incorrect multi-part response encoding by the mock in this PR.

### CODE REVIEW

**Comment by original-brownbear** [comment]
This may also fix https://github.com/elastic/elasticsearch/issues/57115 ... so far I was not able to reproduce randomly closed connections with this fix applied (without it, I can reproduce them relatively quickly).

File: test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java [hunk 1]
```diff
@@ -167,13 +167,13 @@ public void handle(final HttpExchange exchange) throws IOException {
                 final StringBuilder batch = new StringBuilder();
                 for (String line : Streams.readAllLines(requestBody.streamInput())) {
                     if (line.length() == 0 || line.startsWith("--") || line.toLowerCase(Locale.ROOT).startsWith("content")) {
-                        batch.append(line).append('\n');
+                        batch.append(line).append("\r\n");
```
Comment (original-brownbear) [inline test/fixtures/gcs-fixture/src/main/java/fixture/gcs/GoogleCloudStorageHttpHandler.java:170]: This was also broken, I found it in my wireshark adventures during test debugging. We obviously need CLRF in a multi-part response.

**Review by fcofdez — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Francisco! Sadly this didn't fix the test failures after all (looks like those are an SDK bug ...) but at least tests are faster :)

**Pull request merged**